src/actors/objects/mod.rs

```diff
@@ -593,57 +623,86 @@ impl ObjectsActor {
             }))
         });
 
-        file_metas.and_then(move |responses: Vec<Result<Arc<ObjectFileMeta>, _>>|
-                                                 -> Result<Option<Arc<ObjectFileMeta>>, _> {
-            responses
-                .into_iter()
-                .filter(|response| match response {
-                    // Make sure we only return objects which provide the requested info
-                    Ok(object_meta) => {
-                        if object_meta.status == CacheStatus::Positive {
-                            // object_meta.features is meaningless when CacheStatus != Positive
-                            match purpose {
-                                ObjectPurpose::Unwind => object_meta.features.has_unwind_info,
-                                ObjectPurpose::Debug => {
-                                    object_meta.features.has_debug_info
-                                        || object_meta.features.has_symbols
+        file_metas.and_then(
+            move |responses: Vec<Result<Arc<ObjectFileMeta>, _>>| -> Result<FoundObject, _> {
```

Comment:
Is there a way we can get rid of those type annotations? Again, here we can also work with better variable naming, like `object_metas`.

Reply:
`object_meta` is already used as a variable in this code where applicable.  I think `responses` is probably fine-ish.